File: ii-service/src/main/java/com/manymonkeys/service/cinema/KeywordService.java
package com.manymonkeys.service.cinema;

import com.manymonkeys.model.cinema.Keyword;
import com.manymonkeys.service.exception.NotFoundException;

import java.util.List;
import java.util.UUID;

/**
 * @author Ilya Pimenov
 *         Owls Proprietary
 */
public interface KeywordService {

    Keyword createKeyword(String name);

    Keyword loadOrCreateKeyword(String name);

    Keyword loadById(Long id) throws NotFoundException;

    Keyword loadByName(String name) throws NotFoundException;

    List<Keyword> listKeywords();

    Keyword update(Keyword keyword) throws NotFoundException;

}

File: ii-service/src/main/scala/com/manymonkeys/service/cinema/KeywordService.scala
package com.manymonkeys.service.cinema

import com.manymonkeys.model.cinema.Keyword
import java.util.UUID

/**
* @author Ilya Pimenov
*         Owls Proprietary
*/
trait KeywordService {

  def createKeyword(name: String): Keyword

  def loadKeyword(uuid: UUID): Keyword

  def loadKeyword(name: String): Keyword

  def listKeywords: Seq[Keyword]

  def updateName(keyword: Keyword, name: String): Keyword

  def isKeyword(keyword: Keyword): Boolean

}

